kong/plugins.py

```diff
@@ -32,7 +32,52 @@ class Plugins(PluginJsonApply, CrudComponent):
 
 class ServicePlugins(PluginJsonApply, ServiceEntity):
 
+    def wrap(self, data):
+        return Plugin.factory(self, data)
+
     def create(self, skip_error=None, **params):
         params['service_id'] = self.root.id
         return self.execute(self.url, 'post', json=params,
                             wrap=self.wrap, skip_error=skip_error)
+
+
+class Plugin(KongEntity):
+
+    @classmethod
+    def factory(cls, root, data):
+        if data['name'] == 'jwt':
+            return JWTPlugin(root, data)
+        return Plugin(root, data)
+
+
+class JWTPlugin(Plugin):
```

Comment:
do we need this class?
it looks like most operation are via the consumer?

Reply:
They are, but they are still operations from jwt plugin